lib/mozart_fetcher/component.ex

```diff
@@ -16,10 +16,18 @@ defmodule MozartFetcher.Component do
   defp process(_config, {:error, %HTTPoison.Error{reason: reason}}) do
     ExMetrics.increment("error.component.process")
     Stump.log(:error, %{message: "Failed to process HTTP request, reason: #{reason}"})
-    {:error, reason}
+    failed_component(reason)
   end
 
   defp get(config) do
     LocalCache.get_or_store(config.endpoint, fn -> HTTPClient.get(config.endpoint) end)
   end
+
+  defp failed_component(:timeout) do
+    %Component{index: 0, id: "", status: 408, envelope: %Envelope{}}
```

Comment:
Doesn't this mean the component id will be stripped from the debug mode view? It'd be good if we kept the component.id

Reply:
Good spot, I think this could also effect how assembler works out if they were must_succeed or not it will probably look for the ID in the responses

Reply:
created a PR here: https://github.com/bbc/mozart-fetcher/pull/19